source/server/listener_impl.cc

```diff
@@ -126,7 +126,7 @@ Network::SocketSharedPtr ListenSocketFactoryImpl::createListenSocketAndApplyOpti
 
 Network::SocketSharedPtr ListenSocketFactoryImpl::getListenSocket() {
   if (!reuse_port_) {
-    return socket_;
+    return socket_->duplicate();
```

Comment:
Can you add comments here around why duplication is used? Are there any potential threading issues here? I can't remember if this function is called from workers (I think it is), though I'm guessing dup() is safe in a multi-threaded environment?

Reply:
From what I have seen on stackoverflow dup should be safe in multi-threaded. I added a comment to explain why we use dup.

Reply:
added in 8b5dda0f38661892609144ed39df64dec80293cb